io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -332,38 +335,80 @@ schedule !thread@Thread{
       let !t       = NegativeTimeout nextTmid
           !expiry  = d `addTime` time
           !thread' = thread { threadControl = ThreadControl (k t) ctl }
-      trace <- schedule thread' simstate { nextTmid = succ nextTmid }
+      !trace <- schedule thread' simstate { nextTmid = succ nextTmid }
```

Comment:
I don't think we want to bang the result here. It is supposed to yield the trace lazily. That's why we use the lazy ST monad here.

But even if we do want to change it, it should not be done in this commit, since this commit is about something else, and it's better to have one logical change per commit.

Reply:
Done